File: appVendedor/app_vendedor_mipedido/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String description;
  final double price;
  final String imgUrl;
  final List<String> ingredients;
  bool isEnabled;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.imgUrl,
    required this.ingredients,
    this.isEnabled = true,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as String,
      name: json['name'] as String,
      description: json['description'] as String,
      price: (json['price'] as num).toDouble(),
      imgUrl: json['img_url'] as String,
      ingredients: List<String>.from(json['ingredients'] as List),
      isEnabled: json['enabled'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'img_url': imgUrl,
      'ingredients': ingredients,
      'enabled': isEnabled,
    };
  }
}

File: appCliente/app/src/main/java/uk/app02loveslollipop/mipedido/cliente/models/Product.kt
package uk.app02loveslollipop.mipedido.cliente.models

import com.google.gson.annotations.SerializedName

/**
 * Data class representing a product
 */
data class Product(
    @SerializedName("id") val id: String,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String,
    @SerializedName("price") val price: Double,
    @SerializedName("img_url") val imageUrl: String,
    @SerializedName("ingredients") val ingredients: List<String>,
    @SerializedName("isEnabled") val isEnabled: Boolean = true
)